test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py

```diff
@@ -194,31 +194,19 @@ def __get_method_by_name(
 
 
 def __retrieve_is_commissioning(method: FunctionDefType) -> bool:
-    steps_body = __retrieve_return_body(method, ast.List)
-    if not steps_body:
-        return []
-
-    for element in steps_body.value.elts:
-        step = element
-        break
-
-    arg_is_commissioning = False
     try:
-        has_is_commissioning = any(
-            k for k in step.keywords if k.arg == "is_commissioning"
-        )
-        if not has_is_commissioning:
-            return arg_is_commissioning
-
-        arg_is_commissioning = [
-            k.value.value for k in step.keywords if k.arg == "is_commissioning"
+        test_steps = __retrieve_return_body(method, ast.List)
+        first_step = test_steps.value.elts[0]
+        is_commissioning_keyword = [
+            keyword
+            for keyword in first_step.keywords
+            if keyword.arg == "is_commissioning"
         ][0]
-    except Exception as e:
-        logger.error(
-            f"Unable to determine property 'is_commissioning'. Method:{method.name}, Error:{str(e)}"
-        )
 
-    return arg_is_commissioning
+        return is_commissioning_keyword.value.value
+    except Exception as e:
+        logger.error(f"Can't determine 'is_commissioning' value. Error:{str(e)}")
+        return False
```

Comment:
If a test case doesn't have any test steps with the `is_commissioning` parameter then TH will log an error?

Reply:
yes.

Reply:
Please check the refactored code

Reply:
I believe that the problem persists, no?

Reply:
Humm, you are right. I will add an error log line to handle it.